File: backend_spring_boot/src/main/java/com/stage/paiement/dto/RegisterRequest.java
package com.stage.paiement.dto;

import com.stage.paiement.enums.Role;
import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.Setter;

@Getter @Setter
public class RegisterRequest {

    @NotBlank
    private String nomComplet;

    @Email
    @NotBlank
    private String email;

    @Size(min = 6)
    @NotBlank
    private String motDePasse;

    @NotBlank
    private String telephone;

    @NotNull
    private Role role; // CLIENT ou PRESTATAIRE

    // Données spécifiques Prestataire
    private String nomBoutique;
    private String adresseBoutique;
}

File: paiement_facilite_app/lib/models/register_request.dart
class RegisterRequest {
  final String nomComplet;
  final String email;
  final String motDePasse;
  final String telephone;
  final String role;
  final String? nomBoutique;
  final String? adresseBoutique;

  RegisterRequest({
    required this.nomComplet,
    required this.email,
    required this.motDePasse,
    required this.telephone,
    required this.role,
    this.nomBoutique,
    this.adresseBoutique,
  });

  Map<String, dynamic> toJson() {
    return {
      "nomComplet": nomComplet,
      "email": email,
      "motDePasse": motDePasse,
      "telephone": telephone,
      "role": role,
      "nomBoutique": nomBoutique,
      "adresseBoutique": adresseBoutique,
    };
  }
}
